hazelcast/src/test/java/com/hazelcast/quorum/QuorumListenerTest.java

```diff
@@ -69,6 +71,32 @@ public void onChange(QuorumEvent quorumEvent) {
     }
 
     @Test
+    public void testQuorumFailureEventFiredWhenNodeCountBelowThresholdFromClasspathConfiguredListener() throws Exception {
+        Config config = new Config();
+        QuorumListenerConfig listenerConfig = new QuorumListenerConfig();
+        listenerConfig.setClassName("com.hazelcast.quorum.QuorumListenerTest$QuorumListener");
+        String mapName = randomMapName();
+        String quorumName = randomString();
+        QuorumConfig quorumConfig = new QuorumConfig(quorumName, true, 3);
+        quorumConfig.addListenerConfig(listenerConfig);
+        config.getMapConfig(mapName).setQuorumName(quorumName);
+        config.addQuorumConfig(quorumConfig);
+        HazelcastInstance instance = createHazelcastInstance(config);
+        IMap<Object, Object> map = instance.getMap(mapName);
+        try {
+            map.put(generateKeyOwnedBy(instance), 1);
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
I'm a bit unclear. Are you expecting this exception or not? If you are not expecting it, then I would just propagate it and let the test runner deal with the exception.

If you are expecting it, then I would expect something like this:

```
try{
   map.put(...)
   fail();
}catch(TypedException expected){
     ... in most cases you don't want to print the stacktrace.. just noise.
}
```